ddtrace/llmobs/_integrations/utils.py

```diff
@@ -72,3 +75,79 @@ def tag_response_part_google(tag_prefix, span, integration, part, part_idx, cand
         "%s.response.candidates.%d.content.parts.%d.function_call.args" % (tag_prefix, candidate_idx, part_idx),
         integration.trunc(str(_get_attr(function_call, "args", {}))),
     )
+
+
+def llmobs_get_metadata_google(kwargs, instance):
+    metadata = {}
+    model_config = getattr(instance, "_generation_config", {}) or {}
+    model_config = model_config.to_dict() if hasattr(model_config, "to_dict") else model_config
+    request_config = kwargs.get("generation_config", {}) or {}
+    request_config = request_config.to_dict() if hasattr(request_config, "to_dict") else request_config
+
+    parameters = ("temperature", "max_output_tokens", "candidate_count", "top_p", "top_k")
+    for param in parameters:
+        model_config_value = _get_attr(model_config, param, None)
+        request_config_value = _get_attr(request_config, param, None)
+        if model_config_value or request_config_value:
+            metadata[param] = request_config_value or model_config_value
+    return metadata
+
+
+def extract_message_from_part_google(part, role=None):
+    text = _get_attr(part, "text", "")
+    function_call = _get_attr(part, "function_call", None)
+    function_response = _get_attr(part, "function_response", None)
+    message = {"content": text}
+    if role:
+        message["role"] = role
+    if function_call:
+        function_call_dict = function_call
+        if not isinstance(function_call, dict):
+            function_call_dict = type(function_call).to_dict(function_call)
+        message["tool_calls"] = [
+            {"name": function_call_dict.get("name", ""), "arguments": function_call_dict.get("args", {})}
+        ]
+    if function_response:
+        function_response_dict = function_response
+        if not isinstance(function_response, dict):
+            function_response_dict = type(function_response).to_dict(function_response)
+        message["content"] = "[tool result: {}]".format(function_response_dict.get("response", ""))
+    return message
+
+
+def get_llmobs_metrics_tags_google(integration_name, span):
```

Comment:
hey! thats a good suggestion. Looking into the code a bit more, I think it probably makes sense to do this in a follow-up. It won't be as simple as using this helper as-is since some of the other integrations are structured slightly differently. For example, it looks like anthropic uses input / output tokens instead of prompt / completion tokens and bedrock does not have total tokens but instead infers the total from the sum of prompt and completion tokens. All that to say, I think this change deserves a separate PR.